genesyscloud/data_source_genesyscloud_process_automation_trigger.go

```diff
@@ -0,0 +1,100 @@
+package genesyscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v56/platformclientv2"
+	"time"
+	"net/http"
+	"encoding/json"
+	"errors"
+)
+
+func dataSourceProcessAutomationTrigger() *schema.Resource {
+	return &schema.Resource{
+		Description: "Data source for Genesys Cloud process automation trigger. Select a trigger by name",
+		ReadContext: readWithPooledClient(dataSourceProcessAutomationTriggerRead),
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the trigger",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func dataSourceProcessAutomationTriggerRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	sdkConfig := m.(*providerMeta).ClientConfig
+	integrationAPI := platformclientv2.NewIntegrationsApiWithConfig(sdkConfig)
+
+	triggerName := d.Get("name").(string)
+
+	return withRetries(ctx, 15*time.Second, func() *resource.RetryError {
+	    // create path
+    	path := integrationAPI.Configuration.BasePath + "/api/v2/processAutomation/triggers"
+
+		for pageNum := 1; ; pageNum++ {
+			processAutomationTriggers, _, getErr := getAllProcessAutomationTriggers(path, integrationAPI)
+
+			if getErr != nil {
+				return resource.NonRetryableError(fmt.Errorf("failed to get page of process automation triggers: %s", getErr))
+			}
+
+			if processAutomationTriggers.Entities == nil || len(*processAutomationTriggers.Entities) == 0 {
+				return resource.RetryableError(fmt.Errorf("no process automation triggers found with name: %s", triggerName))
+			}
+
+			for _, trigger := range *processAutomationTriggers.Entities {
+				if trigger.Name != nil && *trigger.Name == triggerName {
+					d.SetId(*trigger.Id)
+					return nil
+				}
+			}
+
+			if processAutomationTriggers.NextUri == nil {
+                return resource.NonRetryableError(fmt.Errorf("no process automation triggers found with name: %s", getErr))
```

Comment:
The formatting is off in most files. You can use [go fmt](https://go.dev/blog/gofmt) to format them